### PR TITLE
Vim.Search: Refactor to resolve a TODO

### PR DESCRIPTION
**Pull request by ujihisa**
It has been TODO for 4 to 5 years. Now it's time to resolve! https://github.com/vim-jp/vital.vim/blame/master/autoload/vital/__vital__/Vim/Search.vim#L50

### CODE REVIEW

File: autoload/vital/__vital__/Vim/Search.vim [hunk 1]
```diff
@@ -1,6 +1,15 @@
 let s:save_cpo = &cpo
 set cpo&vim
 
+function! s:_vital_loaded(V) abort
+  let s:V = a:V
```
Comment (tyru) [inline autoload/vital/__vital__/Vim/Search.vim]: no need to define `s:V`

File: autoload/vital/__vital__/Vim/Search.vim [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
+  let s:S = s:V.import('Data.String')
```
Comment (tyru) [inline autoload/vital/__vital__/Vim/Search.vim]: it should use `a:V` instead of `s:V`.

Comment (ujihisa) [reply autoload/vital/__vital__/Vim/Search.vim]: Thanks! Fixed at a1a3bd56b9d966994f5438037f6ffb0e356634ee

**Comment by ujihisa** [comment]
Thx!

めでたい https://twitter.com/ujm/status/1006391434201534464

**Pull request merged**